device/iot-device-samples/send-event/src/main/java/samples/com/microsoft/azure/sdk/iot/SendEvent.java

```diff
@@ -10,233 +10,79 @@
-            }
-            else
-            {
-                System.out.format(
-                        "Expected argument 2 to be one of 'mqtt', 'https', 'amqps' or 'amqps_ws' but received %s\n"
-                                + "The program should be called with the following args: \n"
-                                + "1. [Device connection string] - String containing Hostname, Device Id & Device Key in one of the following formats: HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>\n"
-                                + "2. [number of requests to send]\n"
-                                + "3. (mqtt | https | amqps | amqps_ws | mqtt_ws)\n"
-                                + "4. (optional) path to certificate to enable one-way authentication over ssl for amqps \n",
-                        protocolStr);
-                return;
-            }
-
-            if (args.length == 3)
-            {
-                pathToCertificate = null;
-            }
-            else
-            {
-                pathToCertificate = args[3];
-            }
-        }
-
 
-        System.out.println("Successfully read input parameters.");
-        System.out.format("Using communication protocol %s.\n", protocol.name());
-
-        DeviceClient client = new DeviceClient(connString, protocol);
-        if (pathToCertificate != null )
+        final IotHubEventCallback[] callbacks = new IotHubEventCallback[numberOfMessagesToSend];
+        for (int messageIndex = 0; messageIndex < numberOfMessagesToSend; messageIndex++)
         {
-            client.setOption("SetCertificatePath", pathToCertificate );
+            callbacks[messageIndex] = new EventCallback();
         }
 
-        System.out.println("Successfully created an IoT Hub client.");
-
-        // Set your token expiry time limit here
-        long time = 2400;
-        client.setOption("SetSASTokenExpiryTime", time);
-        System.out.println("Updated token expiry time to " + time);
-
-        client.registerConnectionStatusChangeCallback(new IotHubConnectionStatusChangeCallbackLogger(), new Object());
+        DeviceClient client = new DeviceClient(connString, IotHubClientProtocol.MQTT);
+        long clientSendInterval = 5; //Lower number here spawns send threads more frequently, can send more quickly. By default, value is 10
+        client.setOption("SetSendInterval", clientSendInterval);
+        countDownLatch = new CountDownLatch((int) numberOfMessagesToSend);
 
         client.open();
 
-        System.out.println("Opened connection to IoT Hub.");
-        System.out.println("Sending the following event messages:");
-
-        String deviceId = "MyJavaDevice";
-        double temperature = 0.0;
-        double humidity = 0.0;
-
-        for (int i = 0; i < numRequests; ++i)
+        final long startTime = System.currentTimeMillis();
+        for (int sentMessageCount = 0; sentMessageCount < numberOfMessagesToSend; sentMessageCount++)
```

Comment:
Also this is sequential sending. We need to send it in parallel to execute the right threshholds of iothub

Reply:
i.e change this to use executors

Reply:
done